functests/1_performance/performance.go

```diff
@@ -360,15 +362,26 @@ var _ = Describe("[rfe_id:27368][performance]", func() {
 			By("Checking v1alpha1 -> v1 conversion")
 			v1alpha1Profile.Name = "v1alpha"
 			v1alpha1Profile.ResourceVersion = ""
+			v1alpha1Profile.Spec.NodeSelector = map[string]string{"test": "test"}
 			Expect(testclient.Client.Create(context.TODO(), v1alpha1Profile)).ToNot(HaveOccurred())
-			defer func() { Expect(testclient.Client.Delete(context.TODO(), v1alpha1Profile)).ToNot(HaveOccurred()) }()
 
-			v1Profile := &performancev1.PerformanceProfile{}
 			key = types.NamespacedName{
 				Name:      v1alpha1Profile.Name,
 				Namespace: v1alpha1Profile.Namespace,
 			}
+			defer func() {
+				Expect(testclient.Client.Delete(context.TODO(), v1alpha1Profile)).ToNot(HaveOccurred())
+
+				Eventually(func() bool {
+					if err := testclient.Client.Get(context.TODO(), key, v1alpha1Profile); errors.IsNotFound(err) {
+						klog.Errorf("failed to get performance profile %s: %v", v1alpha1Profile.Name, err)
+						return true
+					}
+					return false
+				}, time.Minute, 10*time.Second).Should(Equal(true))
```

Comment:
A more compact version
	Expect(profiles.WaitForDeletion(key, 60*time.Second)).ToNot(HaveOccurred())
I saw it somewhere else in the code.  I don't mind one wait or another, but maybe we should follow the same conventions.

Reply:
yep but it assumes that you will pass v1 Profile to it and not the v1alpha1

Reply:
I have a patch that fixes also this... in API V2 PR:
https://github.com/openshift-kni/performance-addon-operators/pull/413/commits/3e009043f7558abde7a90621022bef09b0f53315

Reply:
feel free to refactor the test under the API v2 PR :) I want to avoid additional changes under the current PR

Reply:
ok

Reply:
Uhm https://github.com/openshift-kni/performance-addon-operators/commit/3e009043f7558abde7a90621022bef09b0f53315#diff-97cae1e7b24671bf357a6d87a359f597b8b728991d065521ede1c26cc146f73bL52 always expects a v2 Profile, so it doesn't seem a full replacement (yet) of what @cynepco3hahue is doing on his code.

Reply:
@fromanirh it is a full replacement, PAO with a V2 PerformanceProfile expects only  v2 profiles. 
Any non v2 profiles will be converted before served and this is why the above patch is ok (tested)

Reply:
I will rebase the V2 PR aftert his one gets merged and make the necessary changes.

Reply:
ok, will review the new PR.

Reply:
So sad, it already passed CI, never mind.